sensor/common/detector/networkpolicy/networkpolicy.go

```diff
@@ -0,0 +1,48 @@
+package networkpolicy
+
+import (
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/booleanpolicy/augmentedobjs"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/sensor/common/store"
+)
+
+// Finder wraps store.NetworkPolicyStore and provides a convenient method to create
+// augmentedobj.NetworkPoliciesApplied object from a deployment.
+type Finder struct {
+	store store.NetworkPolicyStore
+}
+
+// NewFinder creates a new instance of Finder with provided store.
+func NewFinder(store store.NetworkPolicyStore) *Finder {
+	return &Finder{store: store}
+}
+
+// GetNetworkPoliciesApplied creates an augmentedobj.NetworkPoliciesApplied object
+// based on the provided deployment object and the in-memory store.
+// Finder will use storage.NetworkPolicyType array property on storage.NetworkPolicy proto
+// in order to determine presence or absence of a particular network policy type.
+func (np *Finder) GetNetworkPoliciesApplied(deployment *storage.Deployment) *augmentedobjs.NetworkPoliciesApplied {
+	if !features.NetworkPolicySystemPolicy.Enabled() {
+		// If feature flag is disabled we simply don't do the calculation.
+		// It is fine (from the Matcher perspective) to use nil augmented objects
+		// and no violations will be thrown.
+		return nil
+	}
+
+	var hasIngress, hasEgress bool
+	for _, policy := range np.store.Find(deployment.Namespace, deployment.Labels) {
+		for _, policyType := range policy.GetSpec().GetPolicyTypes() {
+			if policyType == storage.NetworkPolicyType_INGRESS_NETWORK_POLICY_TYPE {
+				hasIngress = true
+			} else if policyType == storage.NetworkPolicyType_EGRESS_NETWORK_POLICY_TYPE {
+				hasEgress = true
+			}
```

Comment:
Maybe some simplification?

```suggestion
			hasIngress = policyType == storage.NetworkPolicyType_INGRESS_NETWORK_POLICY_TYPE
			hasEgress = policyType == storage.NetworkPolicyType_EGRESS_NETWORK_POLICY_TYPE
```

Reply:
We don't want to set `hsaIngress`/`hasEgress` to false ever (apart from the field initialization). With this approach, if the last iterated netpol has no ingress `hasIngress` will be set to `false`, even if it was `true` before.

Reply:
In other words, once it's set to true it should remain true.

Reply:
[nit] Maybe

```go
hasIngress = hasIngress || policyType == storage.NetworkPolicyType_INGRESS_NETWORK_POLICY_TYPE
```

Also I think once `hasIngress` and `hasEgress` are both set to `true` we can early exit the loop.

Reply:
Ah, I see. This is about finding at one matching in a set of potentially many. I thought that there will be always exactly one `NetworkPolicyType_INGRESS_NETWORK_POLICY_TYPE` and one for egress.

Then maybe `hasIngress = hasIngress || policyType == ...` would work for you esthetically?